src/ripple/overlay/impl/OverlayImpl.h

```diff
@@ -389,7 +389,9 @@ class OverlayImpl : public Overlay, public reduce_relay::SquelchHandler
     std::optional<std::uint32_t>
     networkID() const override
     {
-        return setup_.networkID;
+        if (auto const netid = setup_.networkID)
+            return netid;
+        return 0;
```

Comment:
Previously this code could return `std::nullopt` but now it will return an `std::optional` with a value of `0` which (if I understand the code correctly) is the ID of the mainnet. That makes no sense and seems very wrong. Why even have the optional return?

Reply:
I think that this will result in a change of the behavior and I am not sure it is correct. The code previously could return `std::nullopt` and now won't and maybe some parts of the code relied on that? This change should be signed off by someone with better understanding of the codebase than me before you put the change in and admins of servers shoudl be informed.